### PR TITLE
feat: improve reliability of refresh operations

### PR DESCRIPTION
**Pull request by shubha-rajan**

### CODE REVIEW

File: core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java [hunk 1]
```diff
@@ -282,6 +245,60 @@ private static Certificate createCertificate(String cert) throws CertificateExce
     return CertificateFactory.getInstance("X.509").generateCertificate(certStream);
   }
 
+  static void checkDatabaseCompatibility(ConnectSettings instanceMetadata, boolean iamAuth,
```
Comment (enocom) [inline core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:248]: Can we move this organizational cleanup into a separate PR?

**Review by enocom — APPROVED** [review]

File: core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java (hunk 1 above)
Comment (shubha-rajan) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:248]: I use intelliJ to format my code and it's honestly more work to manually undo the formatting fixes we don't want. Can we just leave it for now?

Comment (shubha-rajan) [reply core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java:248]: Actually I could probably make another PR with just the formatting and rebase this one when that's merged.
